presto-druid/src/test/java/io/prestosql/plugin/druid/TestingDruidServer.java

```diff
@@ -169,6 +176,8 @@ public String getHostWorkingDirectory()
     public void close()
     {
         try (Closer closer = Closer.create()) {
+            closer.register(() ->
+                    deleteDirectoryContents(Path.of(hostWorkingDirectory), ALLOW_INSECURE));
```

Comment:
This fails on CI.

Could it be that previous code was failing silently too?

Reply:
Yeah, just saw that as well :(. In the previous code, we were deleting directory contents in the finally block. 
```
finally {
 cleanDirectory(directory)
}
If an exception was thrown, I am hoping that the @AfterClass method in DruidIntegrationSmokeTest calling druidServer.close() would have failed as well.

```
@AfterClass(alwaysRun = true)
    public void destroy()
    {
        if (druidServer != null) {
            druidServer.close();
        }
    }
```

Reply:
Sample successful run with ```cleanDirectory``` - https://github.com/prestosql/presto/runs/783346026

Reply:
i know it was not throwing. But were you actually checking result of file.delete()?

Reply:
Well, locally on my mac, it works fine. I added this assertion:
```
Assert.assertTrue(file.delete(), "Unable to delete: " + file.getName());
```
Going to push by restoring the old code with the assertion added.

Reply:
Yep, fails with the old code too.
```
[ERROR] destroy(io.prestosql.plugin.druid.TestDruidIntegrationSmokeTest)  Time elapsed: 125.035 s  <<< FAILURE!
java.lang.AssertionError: Unable to delete: derby.log expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertTrue(Assert.java:42)
```

Reply:
Can you try finish temporary directory approach (https://github.com/prestosql/presto/pull/3522#discussion_r442144634) ?
Littering in /tmp is not nice, but much better than in a fixed location (tests are not repetitive). -- This would be a intermediate solution.

We could further address cleanup by:

- tweaking the user the druid processes are run as, so that files are removable by us 
- using docker volume to contain the fails, instead of  a host directory